logging/internal/environment.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/compute/metadata"
+)
+
+// ResourceAtttributesGetter abstracts environment lookup methods to query for environment variables, metadata attributes and file content.
+type ResourceAtttributesGetter interface {
```

Comment:
Typo: Extra `t` in `ResourceAtttributesGetter`.

Reply:
Also, I'm wondering if maybe this interface isn't really needed, and maybe you can let `ResourceAttributes` return  defaultResourceGetter instead, and just mock `metadata.Client` in the tests instead? Maybe you're already tried that and it didn't work out, but it looks like it might work to me.